### PR TITLE
feat: integrating ica to rho

### PR DESCRIPTION
**Pull request by yaruwangway**
This PR is for rho:
- [x] add ica controller
- [x] add ica auth (mauth module as a gaia module)
- [ ] add cli to makefile to generate proto (optional)
- [x] complete the todos

### CODE REVIEW

File: x/icamauth/keeper/keeper_test.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+	"github.com/tendermint/tendermint/crypto"
+	"github.com/tendermint/tendermint/libs/log"
+	dbm "github.com/tendermint/tm-db"
+
+	gaiaapp "github.com/cosmos/gaia/v8/app"
+	"github.com/cosmos/gaia/v8/app/helpers"
+)
+
+var (
+	// TODO: Cosmos-SDK ADR-28: Update crypto.AddressHash() when sdk uses address.Module()
```
Comment (yaruwangway) [inline x/icamauth/keeper/keeper_test.go:21]: depends on sdk issue https://github.com/cosmos/cosmos-sdk/issues/10225

**Comment by yaruwangway** [comment]
updates: 
- [ ] ibc team audit
- [ ] https://github.com/cosmos/cosmos-sdk/issues/10225 (optional)
- [x] [tests fail when bumping from ibc v5-beta-1 to v5-rc0](https://github.com/cosmos/interchain-accounts-demo/issues/129) (bump ibc to v5-rc2) will solve this issue (https://github.com/cosmos/ibc-go/pull/2148)

File: app/app.go [hunk 2]
```diff
@@ -959,8 +1002,49 @@ func initParamsKeeper(appCodec codec.BinaryCodec, legacyAmino *codec.LegacyAmino
...
 	paramsKeeper.Subspace(routertypes.ModuleName).WithKeyTable(routertypes.ParamKeyTable())
+	paramsKeeper.Subspace(icacontrollertypes.SubModuleName)
 	paramsKeeper.Subspace(icahosttypes.SubModuleName)
 	paramsKeeper.Subspace(globalfee.ModuleName)
 
 	return paramsKeeper
 }
+
+// todo check
+// func (*App) OnTxSucceeded(ctx sdk.Context, sourcePort, sourceChannel string, txHash []byte, txBytes []byte) {}
```
Comment (glnro) [inline app/app.go]: I would not leave these commented out unimplemented, at minimum log to stderr/out

File: x/icamauth/README.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# ICAMauth module
+
+Interchain account message authentication module (ICAMauth module) is copied from the [interchain-accounts-demo](https://github.com/cosmos/interchain-accounts-demo) respository
+with minor modifications and refator(rename intertx to icamauth). All credits and big thanks go to the original authors.
```
Comment (glnro) [inline x/icamauth/README.md]:
```suggestion
Interchain accounts message authentication module (ICAMauth module) is implemented according to the [interchain-accounts-demo](https://github.com/cosmos/interchain-accounts-demo) repository with some refactoring. Some minor modifications have been made, most notably renaming `intertx` to `icamauth`. All credit  and thanks go to the original authors.
```

File: x/icamauth/keeper/grpc_query.go [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	icatypes "github.com/cosmos/ibc-go/v5/modules/apps/27-interchain-accounts/types"
+
+	"github.com/cosmos/gaia/v8/x/icamauth/types"
+)
+
+// InterchainAccountFromAddress implements the Query/InterchainAccount gRPC method
```
Comment (okwme) [inline x/icamauth/keeper/grpc_query.go]:
```suggestion
// InterchainAccount implements the Query/InterchainAccount gRPC method
```

File: app/app.go [hunk 5]
```diff
@@ -175,6 +182,8 @@ var (
 		govtypes.ModuleName:            {authtypes.Burner},
 		liquiditytypes.ModuleName:      {authtypes.Minter, authtypes.Burner},
 		ibctransfertypes.ModuleName:    {authtypes.Minter, authtypes.Burner},
+		// todo: check here from interchain-acount-demo
```
Comment (okwme) [inline app/app.go]: looks the same as https://github.com/cosmos/interchain-accounts-demo/blob/master/app/app.go#L163
besides that they include ibcfee module and we do not so both of these commented out lines could be removed.

**Review by okwme — APPROVED** [review]
👍

**Pull request merged**